functorch/test/test_control_flow.py

```diff
@@ -0,0 +1,154 @@
+        def true_nested(y):
+            return y * y
+
+        def false_nested(y):
+            return y + y
+
+        def true_fn(x, pred2):
+            z = cond(pred2, true_nested, false_nested, x)
+            return x + z
+
+        def false_fn(x, _):
+            return x.cos()
+
+        def f(x, pred, pred2):
+            return cond(pred, true_fn, false_fn, (x, pred2))
+
+        x = torch.randn(4)
+        graph = make_fx(f)(x, torch.tensor(False), torch.tensor(False))
+
+        result_true_true = graph.forward(x, torch.tensor(True), torch.tensor(True))  # True + True -> x * x
+        result_true_false = graph.forward(x, torch.tensor(True), torch.tensor(False))  # True + True -> x + x
+        result_false_true = graph.forward(x, torch.tensor(False), torch.tensor(True))  #  False + either -> cos
+        result_false_false = graph.forward(x, torch.tensor(False), torch.tensor(False))  #  False + either -> cos
+
+        self.assertNotEqual(result_true_true, result_true_false)
+        self.assertFalse(torch.allclose(result_false_true, result_true_true))
+
+        self.assertTrue(torch.allclose(result_false_true, result_false_false))
+
+        self.assertTrue(torch.allclose(result_true_true, (x * x) + x))
+        self.assertTrue(torch.allclose(result_true_false, x + x + x))
+
+        self.assertTrue(torch.allclose(result_false_true, torch.cos(x)))
+
+    def test_cond_nested_traced_other_inputs(self):
+        def true_nested(y):
+            return y * y
+
+        def false_nested(y):
+            return y + y
+
+        def true_fn(k, pred2):
+            z = cond(pred2, true_nested, false_nested, k)
+            return torch.add(torch.tensor([.25, .25]), z)
+
+        def false_fn(k, _):
+            return k.cos()
+
+        def f(k, pred, pred2):
+            return cond(pred, true_fn, false_fn, (k, pred2))
+
+        x = torch.tensor([0.5, 0.5])
+        graph = make_fx(f)(x, torch.tensor(False), torch.tensor(False))
+
+        a = torch.tensor([1.0, 1.0])
+        result_true_true = graph.forward(a, torch.tensor(True), torch.tensor(True))
+        self.assertTrue(torch.allclose(result_true_true, (a * a) + torch.tensor([0.25, 0.25])))
+
+        b = torch.tensor([2.0, 2.0])
+        result_true_true = graph.forward(b, torch.tensor(True), torch.tensor(True))
+        self.assertTrue(torch.allclose(result_true_true, (b * b) + torch.tensor([0.25, 0.25])))
+
+    def test_cond_nested_traced_multi(self):
+        def true_a(y):
+            return y * y
+
+        def false_a(y):
+            return y + y
+
+        def true_b(y, z):
+            return y + z
+
+        def false_b(y, z):
+            return y * z
+
+        def f(x, pred, pred2):
+            a_out = cond(pred, true_a, false_a, (x))
+            b_out = cond(pred2, true_b, false_b, (x, x))
+            return a_out + b_out
+
+        x = torch.randn(4)
+        graph = make_fx(f)(x, torch.tensor(False), torch.tensor(False))
```

Comment:
Maybe more of a question for @Chillee @eellison -- do we want to test that this works under the other flavors of make_fx (I'm mostly just thinking fake Tensor tracing; this is probably not going to work with symbolic yet) or do we punt that to the future?